compiler/noirc_evaluator/src/ssa/ir/function.rs

```diff
@@ -169,11 +169,12 @@ impl Function {
     /// want to iterate only reachable blocks.
     pub(crate) fn reachable_blocks(&self) -> BTreeSet<BasicBlockId> {
         let mut blocks = BTreeSet::new();
-        let mut stack = vec![self.entry_block];
+        let mut deque = VecDeque::new();
+        deque.push_back(self.entry_block);
 
-        while let Some(block) = stack.pop() {
+        while let Some(block) = deque.pop_front() {
```

Comment:
So this would do a BFS instead of the DFS that the display originally did, however it's all defeated by the fact that `BTreeSet` will order by numeric ID.

Reply:
Oh, now I understand. I missed how `BTreeSet` works

Reply:
I think the insertion-order sensitive version is https://docs.rs/indexmap/latest/indexmap/
I came across it yesterday in the optimiser